docs/source/newsfragments/4093.removal.rst

```diff
@@ -0,0 +1 @@
+When constructing a :class:`~cocotb.types.LogicArray` from a :class:`int`, the *range* argument is now required, as guesses are rarely correct or useful.
```

Comment:
I wouldn't say the int -> LogicArray is neither correct nor useful. I think the behavior we have right now is probably what users want in many cases. I'm fine with removing it, though (and maybe add it back later if users *really* want it), but we should at least help them getting the code right.

For that, can you add guidance for users how to get the old behavior back? E.g. "Use `max(1, int.bit_length(value))` to calculate the minimum size of the array for a given integer `value`."

Reply:
If you are converting an *unknown* integer value, and then doing the following...
1. Assigning to a handle eventually. Handle assignment requires exactly sized argument, so depending upon guessing won't work.
2. Comparison to another LogicArray. LogicArray == LogicArray requires them to be the same size, so depending upon guessing won't work.
3. Bitwise operations with another LogicArray. This too requires two LogicArrays of the same size, so depending upon guessing won't work.

It only works if someone *knows* the value, thus *knows* how big they want it, and *knows* how LogicArray guesses size, but lazily does not specify it. This is gaming the system in a way that makes test more fragile and less readable. If you know how big it's supposed to be, let us know too...

That's at least what was going through my head when I decided to remove it. I'll remove the value judgement from the newsfrag tho.
```suggestion
When constructing a :class:`~cocotb.types.LogicArray` from a :class:`int`, the *range* argument is now required.
```